### PR TITLE
Post-pex simulations, organized simulation options, new notebook

### PR DESCRIPTION
**Pull request by alibillalhammoud**
This update adds the post PEX simulations and brings simulations together in a more organized fashion
I also added the developed notebook.

credits to:
@Sai provided some starter commands and helped me with edits to get the python env working correctly
@Lucca pointed me to the commands needed to get a gds dump from the database files from openroad
@Qirui provided silicon data

several updates:
simulations
	rewrite update netlist so that the "partial" simulation functionality is included for both prePEX and postPEX netlists
	add match function to find common names that should be removed from partial netlist
	substitute a standard toplvl definition
	specify optional prePEX vs postPEX simulations or both from the command line when calling the generator 
		also edit makefile to include the new command line options

notebook
	in docs/source the tempsense notebook is now further developed
	The notebook explains the temperature sensor and flow aimed at a non-expert user
	includes interactive components to visualize the flow progress and see simulation results
	stil does not include simulations (will update later today)
	uses klayout deb package

other
	Qirui's PR provided silicon data; move this into temp sense gen directory and adapt into "modelfile" (keep the A-hd data)
	add prePEX model file and add postPEX model file
	each simulation run type gets its own directory and output

### CODE REVIEW

**Comment by alibillalhammoud** [comment]
I will run pre-commit later today.
Also please suggest notebook improvements to what I have done so far, or suggest corrections if anything appears incorrect. 
To clarify every notebook command on this PR should run to completion, so if a command fails, let me know. The notebook runs the entire flow and prints step by step graphical outputs.

File: docs/source/notebookFiles/def2gds.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import copy
```
Comment (msaligane) [inline docs/source/notebookFiles/def2gds.py:1]: These files are already in the sensor's directory. Why don't you point to them?
Same for the gds files above.

File: docs/source/notebookFiles/fill.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{
```
Comment (msaligane) [inline docs/source/notebookFiles/fill.json:1]: same as above

File: docs/source/temp_sense_genCollab.ipynb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
...zwgKtjmxaxpTotOZ8tbzHP8W233Zbqx+L8y9YzQltX1ly3XRmzGpeM0R+w+OmkHVP+U+UsNWpyjowz2WwytrmQlLcUvrERBpcdE+OAvdzSw5YdS6MnLD1y0u+FF17YaDMtqycLua2+RAgIASEgBISAEBACQkAICIHKCGDdtNSoCemVKYvuOx9TplxlrK8wr99///0u68kPfvCD6IknnnDWYgi3LrnkEmflfeuttyJT2iOYscsJWT5MuY+OPfbY6He/+11kylpkaUudZRXLLdZh4oGxkJ966qnR448/Hv3+97+PzjjjDGdFNg0qMqXbjSlPsMpaOtiIeVgaq+jTTz+N/vrXv0ZnnnlmRNwwmVGwxkPqFQrWYKy8pgS7w3vttVdkirer/+GHH0amADoWbv6/9dZbl1i8fVum0Ea33357tPzyy0dY7WkHqzeZbpAjjzwyiV2eMGGCw+Cjjz6KTLl1mGM1NyXYzQ8rO6zqXh555JGUh4CloU3WCvyKFgj+wJxxMhes+d5CDxkaHguzZ89218QPf/hDd03YpobzrIAYEIv/m2++6a6Jd955p9DhcV14yWNRL7SzRhprmaqvhoWAEBACQkAICAEhIASEgBDoOQRMoXMWR1PCSuZmymxVa+Qrr7wSewuupQRNtRFayOnjrrvuKunDFPSkD1MCY1NOS8ocf/zxSZnHHnssdd5byGnf3M1jU+5L6k+bNi2pb+zcqfO/+MUvknOnnHJKSV0O2EZDbG7ortwmm2ySKuMt5JzbeOONnSXcix+LKd6xbQq4+lji88ZIHVPEk7E8/fTTqX5og/p8zD09dY4vRVrIv/KVr8Sm+Kb68GO2zZJkHJdeemnJODhgGymxv67GjRuXW6bRgxaO4Pqn/c8//7zRZlpWTxZyWx2JEBACQkAICAEhIASEgBAQAs0hYBpLdMEFF7hG1ltvPWcJzRNzfXcp0xAsxOUsolgzsYBnBSuqF3P5dlblrJgSnBzKxrf7E7Bymwuzs9pmBevu6quv7g5nY6F9+iwIwrDe5wlx81i4kYceeih64YUX8opFtnHgLP5e/FjwKCD3NTHaxMfnjZE6m2++eVIXC327ZOzYsS5uPBTGjCcDGCOcx5KfJ6uuuqqz+CN4YsAfUIT885//dF4TyLBhwyLbOCii2ULbkEJeKJxqTAgIASEgBISAEBACQkAI9CcCkJ
```
Comment (saicharan0112) [inline docs/source/temp_sense_genCollab.ipynb:10]: 1. Add a separate cell in the beginning of the notebook as a header for the sphinx to include it in docs (Don't edit here, you might miss the syntax. Edit with the notebook running in the browser.) Something like below - 
image
2. Also run all the cells and add that copy of notebook to the docs. Seeing the result of each cell will be really good in the docs.

**Review by saicharan0112 — CHANGES_REQUESTED** [review]
Rename the notebook in the last line of /docs/source/index.rst to this new notebook.
OpenFASoC_temp_sense_gen -> temp_sense_genCollab

**Comment by saicharan0112** [comment]
@alibillalhammoud let me know once you are done with the changes regarding the notebook

**Comment by luccareinehr** [comment]
@alibillalhammoud looks good! Could you squash some of the commits to ease the history?

**Review by luccareinehr — APPROVED** [review]
If CI succeeds, looks good

**Comment by luccareinehr** [comment]
@alibillalhammoud I see most of the files from `docs/source/notebookFiles/` look like copies of files present in other directories in openfasoc. Would it be possible to delete this folder altogether and just fetch the original files in your notebook code?

This way we can avoid breaking the notebook if the flow gets updated, and also reduce the size of the repo (that is already quite large).

**Comment by saicharan0112** [comment]
@alibillalhammoud You missed two more changes I suggested. This removed the notebook from the docs. Also the cells are not executed.
@msaligane are you not planning to display the notebook at openfasoc.readthedocs.io?

**Comment by msaligane** [comment]
Yes, I missed that. We need the colab ready tonight. We can add fixes to those.
@saicharan0112 can you take care of that @alibillalhammoud is busy the next 2 days and I am working on the formatting of the colab.

**Comment by saicharan0112** [comment]
Sure. I will add that in lucca's PR.

**Pull request merged**